Completion of Lab 1 Tasks

kiphenglim opened the pull request:
We have completed the HTML/CSS pages that correspond to the tasks outlined in the Lab 1 writeup, as well as a design document that details all of our choices and example screenshots.

nrhodes-hmc commented on lab1/design.md:
```diff
@@ -0,0 +1,55 @@
+# Design Decisions
++ used flexbox for list items for better and easier
```
This is an implementation decision, not a design decision.

Design is how it looks, implementation is how it was achieved.

kiphenglim commented:
> The design.md doesn't show "the final design, including screen images and the flow for each task"

We added lots of screenshots!

> In https://www.kiphenglim.com/fa21-cs124/lab1/t1-begin.html, there are no completed items. Should "Remove completed" be there?

Probably not, but we added all the screenshots and were too lazy to change it.

> I can't tell what the process is for adding an item. Do you type in the edit box and then click Add. If so, you need to show/describe that.

That's reasonable. We added screenshots and a text description for how to add an item.

> I like the strikeout of completed items (and the dimming of the text). I assume the item is completed by clicking on the checkbox?

Yes, the user is supposed to just click the checkbox.

> https://www.kiphenglim.com/fa21-cs124/lab1/t4-edit.html: Again, don't make me guess as to what user actions are causing the change from one screen to the next. That's what should be in the design doc.

Fair enough; we added more detail as screenshots and text descriptions as well.

> When editing an item's description, what happens if I make changes, and then want to revert? Is there a way to cancel?

There's no way to revert your changes, per se. Whatever is in the text box when you click "Done" is what you get

> I assume I'm allowed to have multiple items in the process of being edited at the same time?

There is nothing stopping you from editing multiple items.

> Why have an edit button at all? Why not just click on the text and start editing?

That's a great point. If we were to have another iteration of this app, we would implement that idea.
 
> Commonly on mobile apps, buttons are at the bottom of the screen.

Noted. We'll keep this in mind for lab 2.

> If I accidentally click on Remove Completed, is there any way to undo (or confirmation dialog)?

No undo or confirmation. We're being mean to our hypothetical users for the sake of ease of implementation.

nrhodes-hmc commented:
Missing GitHub pages

kiphenglim commented:
Sorry! Forgot you hadn’t graded lab1 yet and we deployed our lab2 code :,)
